### PR TITLE
Add sortStatus renderer

### PR DESCRIPTION
**Pull request by amanmahajan7**
Fixes https://github.com/adazzle/react-data-grid/issues/3000

This replaces `sortIcon` renderer. Now both icon and priority can be customized using the new `sortStatus` renderer

### CODE REVIEW

**Review by nstepien — DISMISSED** [review]
I'm fine with the changes, but I wonder if it'd be better if we could render both at once instead, or if we could customize header cells as a whole.

**Comment by amanmahajan7** [comment]
Wdyt about adding `priority` prop to `sortIcon`? We will have to re-export `sortIcon` I think

**Comment by nstepien** [comment]
I don't mind re-exporting `sortIcon`.
I think rather than adding `priority` to `sortIcon`, we can add `sortStatus` that renders both `sortIcon` and `sortPriority`

File: src/types.ts [hunk 1]
```diff
@@ -213,6 +213,12 @@ export interface SortIconProps {
   sortDirection: SortDirection | undefined;
 }
 
+export interface SortPriorityProps {
+  priority: number | undefined;
+}
+
+export type SortStatusProps = SortIconProps & SortPriorityProps;
```
Comment (nstepien) [inline src/types.ts:220]: I believe this is better for 1. performance, 2. catching conflicting types
```suggestion
export interface SortStatusProps extends SortIconProps, SortPriorityProps {};
```

**Review by nstepien — APPROVED** [review]

**Review by nstepien — APPROVED** [review]

**Review by nstepien — APPROVED** [review]

**Pull request merged**